### PR TITLE
Improve waiting behavior

### PR DESCRIPTION
**Pull request by pkmiec**
There are 2 main problems with AePageObjects making it prone to flakiness,

- The page polling in AePageObjects uses `Capybara.using_wait_time(0)` preventing Capybara from retrying errors and reloading nodes. The page polling also does not catch any errors itself causing exceptions to be potentially raised prior to waiting the asked for time. The page polling is used `browser.find_document`, `window.change_to`, and `absent?` / `present?` / and friends on element proxy.

- The `ensure_loaded!` is difficult to write / reason about as it is sometimes called inside of page polling block (see above) and sometimes it is not. This means that implementors have to know about all the exceptions that should be caught and using `AePageObjects#wait_until` is not good because it can blocks for longer than what is intended by page polling. 

This PR improves AePageObjects by,

- Removing PagePolling and instead solely relying on the improved `#wait_until`. It now rescues and retries all relevant selenium / capybara / ae_page_objects exceptions. It only throws 'WaitTimeoutError' (which includes the original cause) making it easy for user to rescue errors. It lowers Capybara's default time allowing capybara to retry / reload, but also allowing errors to be handled at the higher level (and thus another chance to success). It allows itself to be called multiple times which the top-level invocation being in charge of the timeout.

- The `ensure_loaded!` now calls `is_loaded?` which is meant to quickly return an answer. Subclasses should override `is_loaded?` now, but no waiting / rescuing is needed.

- Removed `#all` / `#find` from being delegated to Capybara node. These return Capybara nodes (and in case of all not reloadable once) and it is better to be explicit about their use. Also avoids confusion with `#find` provided by `AePageObjects::Colleciton`.

### CODE REVIEW

File: lib/ae_page_objects/util/page_polling.rb [hunk 1]
```diff
@@ -1,53 +0,0 @@
-module AePageObjects
```
Comment (tlconnor) [inline lib/ae_page_objects/util/page_polling.rb]: Are these file removals meant to be part of this commit or the next?

File: lib/ae_page_objects.rb [hunk 2]
```diff
@@ -35,22 +32,80 @@ def browser
       end
     end
 
-    def wait_until(seconds_to_wait = nil, error_message = nil)
-      seconds_to_wait ||= default_max_wait_time
-      start_time      = Time.now
+    def wait_until(seconds_to_wait = nil, error_message = nil, &block)
```
Comment (tlconnor) [inline lib/ae_page_objects.rb]: You've called this commit `pk - passing tests` but really this one contains the bulk of the changes.

File: lib/ae_page_objects.rb [hunk 3]
```diff
@@ -35,22 +32,80 @@ def browser
...
+      @wait_until ||= 0
+      @wait_until += 1
+      # start_time = Time.now
```
Comment (tlconnor) [inline lib/ae_page_objects.rb]: Are these commented lines needed anymore?

File: lib/ae_page_objects.rb [hunk 4]
```diff
@@ -35,22 +32,80 @@ def browser
...
+          sleep(0.05)
+          raise FrozenInTime, "Time appears to be frozen" if Time.now == start_time
+
+          retry
+        end
       end
 
       result
+    ensure
+      # puts "#{"--" * @wait_until} wait_until => #{Time.now - start_time}"
```
Comment (tlconnor) [inline lib/ae_page_objects.rb]: Commented code

File: lib/ae_page_objects/version.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module AePageObjects
-  VERSION = '2.1.0'.freeze
+  VERSION = '2.0.1'.freeze
```
Comment (tlconnor) [inline lib/ae_page_objects/version.rb]: Why was the version number decreased?  Given the API changes we'll need a major version number, making this `3.0.0`

File: test/test_apps/shared/test/selenium_helper.rb [hunk 6]
```diff
@@ -1,5 +1,6 @@
 ENV['RAILS_ENV'] = 'test'
 
+puts $LOAD_PATH
```
Comment (tlconnor) [inline test/test_apps/shared/test/selenium_helper.rb]: Might want to remove this 😃

File: lib/ae_page_objects.rb (hunk 2 above)
Comment (pkmiec) [inline lib/ae_page_objects.rb]: Reworded the commit msg.

File: lib/ae_page_objects/util/page_polling.rb (hunk 1 above)
Comment (pkmiec) [inline lib/ae_page_objects/util/page_polling.rb]: Yes, that got messed up a bit. I squashed most of it together now with a better msg.

File: lib/ae_page_objects.rb (hunk 4 above)
Comment (pkmiec) [inline lib/ae_page_objects.rb]: Removed

File: lib/ae_page_objects.rb (hunk 3 above)
Comment (pkmiec) [inline lib/ae_page_objects.rb]: Removed.

**Pull request merged**